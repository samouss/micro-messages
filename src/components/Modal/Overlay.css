.Overlay {
  visibility: hidden;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.75);
  opacity: 0;
  transition: visibility 0s 0.25s, opacity 0.25s;
}

.Overlay--visible {
  opacity: 1;
  visibility: visible;
  transition-delay: 0s;
}
