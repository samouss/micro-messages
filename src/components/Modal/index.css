.Modal {
  position: absolute;
  z-index: 1000;
  visibility: hidden;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.Modal__Container {
  background: #FFFFFF;
  width: 50%;
  min-width: 250px;
  max-width: 500px;
  border-radius: 2px;
  visibility: hidden;
  opacity: 0;
  transform: translate3d(0, 100%, 0);
  -webkit-backface-visibility: hidden;
  -webkit-perspective: 1000;
  transition:
    visibility 0s 0.25s,
    transform 0.25s,
    opacity 0.25s;
}

.Modal__Container--visible {
  visibility: visible;
  opacity: 1;
  transform: translateY(0);
  transition-delay: 0s;
}

.Modal__Header {
  padding: 10px 10px 0;
  text-align: right;
}

.Modal__Header__Button {
  background: none;
  border: none;
  padding: 10px;
  transform: rotate(45deg);
}

.Modal__Header__Close:before,
.Modal__Header__Close:after {
  width: 15px;
  background-color: #000000;
}

.Modal__Content {
  padding: 25px;
}
