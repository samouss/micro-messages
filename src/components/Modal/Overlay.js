// @flow

import React from 'react';
import cx from 'classnames';
import './Overlay.css';

export type Props = {
  isOpen: boolean,
};

const Overlay = ({ isOpen }: Props) => (
  <div
    styleName={cx('Overlay', {
      'Overlay--visible': isOpen,
    })}
  />
);

export default Overlay;
