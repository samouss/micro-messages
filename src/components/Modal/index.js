// @flow

import React from 'react';
import cx from 'classnames';
import Plus from 'components/Plus';
import Overlay from './Overlay';
import styles from './index.css';

export type Props = {
  isOpen: boolean,
  onClose: () => void,
  children: React.Element<*>,
};

const Modal = ({ isOpen, onClose, children }: Props) => (
  <div styleName="Modal">
    <Overlay
      isOpen={isOpen}
    />

    <div
      styleName={cx('Modal__Container', {
        'Modal__Container--visible': isOpen,
      })}
    >
      <div styleName="Modal__Header">
        <button
          styleName="Modal__Header__Button"
          onClick={onClose}
        >
          <Plus
            customClassName={styles.Modal__Header__Close}
          />
        </button>
      </div>

      <div styleName="Modal__Content">
        {children}
      </div>
    </div>
  </div>
);

export default Modal;
