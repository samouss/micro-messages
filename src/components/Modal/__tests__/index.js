// @flow

import React from 'react';
import { shallow } from 'enzyme';
import Modal from '../index';

describe('<Modal />', () => {
  const defaultProps = {
    isOpen: false,
    onClose: () => {},
  };

  it('expect to render the component correctly closed', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <Modal
        {...props}
      >
        <div>some content</div>
      </Modal>,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render the component correctly open', () => {
    const props = {
      ...defaultProps,
      isOpen: true,
    };

    const component = shallow(
      <Modal
        {...props}
      >
        <div>some content</div>
      </Modal>,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to call onClose when button is clicked', () => {
    const props = {
      ...defaultProps,
      onClose: jest.fn(),
    };

    const component = shallow(
      <Modal
        {...props}
      >
        <div>some content</div>
      </Modal>,
    );

    component
      .find('[styleName="Modal__Header__Button"]')
      .simulate('click');

    expect(props.onClose).toHaveBeenCalled();
  });
});
