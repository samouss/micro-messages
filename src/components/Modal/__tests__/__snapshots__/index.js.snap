// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> expect to render the component correctly closed 1`] = `
<div
  styleName="Modal"
>
  <Overlay
    isOpen={false}
  />
  <div
    styleName="Modal__Container"
  >
    <div
      styleName="Modal__Header"
    >
      <button
        onClick={[Function]}
        styleName="Modal__Header__Button"
      >
        <Plus
          customClassName="Modal__Header__Close"
        />
      </button>
    </div>
    <div
      styleName="Modal__Content"
    >
      <div>
        some content
      </div>
    </div>
  </div>
</div>
`;

exports[`<Modal /> expect to render the component correctly open 1`] = `
<div
  styleName="Modal"
>
  <Overlay
    isOpen={true}
  />
  <div
    styleName="Modal__Container Modal__Container--visible"
  >
    <div
      styleName="Modal__Header"
    >
      <button
        onClick={[Function]}
        styleName="Modal__Header__Button"
      >
        <Plus
          customClassName="Modal__Header__Close"
        />
      </button>
    </div>
    <div
      styleName="Modal__Content"
    >
      <div>
        some content
      </div>
    </div>
  </div>
</div>
`;
